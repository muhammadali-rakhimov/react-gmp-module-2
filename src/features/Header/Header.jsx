/* eslint-disable import/no-named-as-default-member */
/* eslint-disable import/no-named-as-default */
/* eslint-disable react/jsx-no-bind */
/* eslint-disable jsx-a11y/no-static-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React, { useState, useEffect } from 'react'
import { connect, useDispatch } from 'react-redux'
import clsx from 'clsx'
import { Row, Container, Col } from 'react-bootstrap'
// import { useNavigate, useParams } from 'react-router-dom'
import {
  getSelectedMovieNull,
  searchMoviesAmount,
  setSearchInput,
  setSearchMovies,
} from '../../redux/actions/actions'
import styled from './Header.module.scss'
import wallpaper from '../../../public/img/header-background.png'
import AddMovieModal from '../AddMovieModal/AddMovieModal'
import toUpper from '../../shared/utils/toUpper'
import Modal from '../../base/Modal/Modal'
import { useGetSearchMoviesQuery } from '../../services/api'

function Header(
  {
    selectedMovie,
    selectedMovieNull,
    searchMovies,
    getSearch,
    getSortBy,
    setSearchMoviesAmount,
  },
  page
) {
  const [isOpen, setIsOpen] = useState(false)

  const dispatch = useDispatch()

  const { data, isError, error } = useGetSearchMoviesQuery({
    search: getSearch,
    sortBy: getSortBy,
  })

  useEffect(() => {
    if (isError) {
      console.log(error)
    }
  }, [isError, error])

  // const navigate = useNavigate()

  return (
    <Container>
      <Row>
        <nav
          style={{ backgroundImage: `url(${wallpaper})` }}
          className={clsx(styled.parent, 'navbar', 'navbar-light', 'p-5')}
        >
          <div className={clsx('container-fluid', styled.containerHeader)}>
            <a href="/" className={styled.logo}>
              <b>netflix</b>
              roulette
            </a>
            <div>
              {selectedMovie === null ? (
                <button
                  onClick={() => setIsOpen(true)}
                  className={styled.addMovie}
                  type="button"
                >
                  <span>+ </span>
                  {toUpper('Add Movie')}
                </button>
              ) : (
                <div
                  onClick={() => dispatch(selectedMovieNull())}
                  className={styled.cursor}
                >
                  <svg
                    width="25"
                    height="26"
                    viewBox="0 0 29 30"
                    fill="none"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="18.5"
                      cy="10.5"
                      r="9.5"
                      stroke="#F65261"
                      strokeWidth="2"
                    />
                    <path
                      d="M10.5 19.5L1.5 28.5"
                      stroke="#F65261"
                      strokeWidth="2"
                      strokeLinecap="square"
                    />
                  </svg>
                </div>
              )}
            </div>
          </div>

          <Modal handleClose={() => setIsOpen(false)} isOpen={isOpen}>
            <AddMovieModal handleClose={() => setIsOpen(false)} />
          </Modal>

          {selectedMovie === null ? (
            <div className={clsx(styled.find, 'col', 'p-5')}>
              <div className="container">
                <h1>FIND YOUR MOVIE</h1>
                <br />
                <form onSubmit={(e) => e.preventDefault()}>
                  <input
                    type="text"
                    className={`${styled.search} col col-md-9 p-3`}
                    placeholder="What do you want to watch?"
                    onChange={(e) => dispatch(setSearchInput(e.target.value))}
                    value={getSearch}
                  />
                  <button
                    onClick={() => {
                      // navigate(
                      //   `/search/${getSearch}?genre=all&sortBy=${getSortBy}&sortOrder=desc`,
                      //   {
                      //     state: {
                      //       page,
                      //     },
                      //   }
                      // )
                      dispatch(searchMovies(data))
                      dispatch(setSearchMoviesAmount(data.totalAmount))
                    }}
                    className={`${styled.searchButton} col col-md-3 p-3`}
                    type="submit"
                  >
                    SEARCH
                  </button>
                </form>
              </div>
            </div>
          ) : (
            <div className="container-fluid my-5 text-light">
              <Row>
                <Col className={clsx(styled.parentCard, 'd-flex')}>
                  <img
                    className={clsx(
                      selectedMovie === null ? styled.image : styled.imageFocus,
                      ''
                    )}
                    src={selectedMovie.poster_path}
                    alt="img"
                  />
                  <div className="ms-4">
                    <p className={styled.movieName}>
                      {selectedMovie.title}
                      <span className={styled.rating}>
                        {selectedMovie.vote_average
                          ? selectedMovie.vote_average
                          : '0.0'}
                      </span>
                    </p>
                    <div className={styled.movieType}>
                      {selectedMovie.genres
                        ? selectedMovie.genres.map((item, index) => (
                            // eslint-disable-next-line react/jsx-indent
                            <span key={index}>{`${item} `}</span>
                            // eslint-disable-next-line indent
                          ))
                        : null}
                    </div>
                    <div className={clsx(styled.yearDuration, 'd-flex')}>
                      <p className={styled.year}>
                        {selectedMovie.release_date
                          ? selectedMovie.release_date
                          : 'unknown'}
                      </p>
                      <p className={styled.duration}>
                        {`${Math.trunc(selectedMovie.runtime / 60)}h ${
                          selectedMovie.runtime -
                          Math.trunc(selectedMovie.runtime / 60) * 60
                        }min`}
                      </p>
                    </div>
                    <p>
                      {selectedMovie.overview
                        ? selectedMovie.overview
                        : 'Unknown overview'}
                    </p>
                  </div>
                </Col>
              </Row>
            </div>
          )}
        </nav>
      </Row>
    </Container>
  )
}

const mapStateToProps = (state) => {
  return {
    selectedMovie: state.root.selectedMovie,
    getSearch: state.root.search,
    getSortBy: state.root.sortBy,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    selectedMovieNull: () => dispatch(getSelectedMovieNull()),
    searchMovies: (item) => dispatch(setSearchMovies(item)),
    setSearchMoviesAmount: (number) => dispatch(searchMoviesAmount(number)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Header)
