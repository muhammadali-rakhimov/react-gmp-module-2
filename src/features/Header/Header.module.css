@import url('https://fonts.googleapis.com/css2?family=Montserrat&display=swap');
.parent {
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-size: cover;
  background-color: #000000;
}

.parent .containerHeader .addMovie {
  font-family: 'Montserrat', sans-serif;
  font-size: 18px;
  color: #f65261;
  background: rgba(96, 96, 96, 0.68);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  padding: 8px;
  outline: 0;
}

.parent .containerHeader .addMovie:active {
  background: rgba(74, 74, 74, 0.68);
}

.parent .containerHeader a {
  text-decoration: none;
  font-family: 'Montserrat', sans-serif;
  font-size: 25px;
  color: #f65261;
}

.parent .find {
  color: #ffffff;
  font-family: 'Montserrat', sans-serif;
}

.parent .search {
  outline: none;
  border: 0;
  border-radius: 5px;
  margin-right: 10px;
  width: 70%;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: #424242;
  color: #ffffff;
}

@media screen and (max-width: 992px) {
  .parent .imageFocus {
    width: 100%;
    -o-object-fit: contain;
    object-fit: contain;
  }
  .parent .parentCard {
    display: block !important;
  }
  .parent .parentCard img {
    margin-bottom: 40px;
  }
}

.parent .parentCard {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.parent .parentCard .image {
  width: 100%;
}

.parent .parentCard .movieName {
  font-size: 40px;
}

.parent .parentCard .rating {
  border: 1px solid #ffffff;
  border-radius: 100%;
  padding: 5px 10px 10px 10px;
  font-size: 35px;
  margin-left: 15px;
  color: #98e287;
}

.parent .parentCard .movieType {
  font-size: 20px;
  color: #cccccc;
}

.parent .parentCard .parentCard {
  font-family: 'Montserrat';
}

.parent .parentCard .yearDuration {
  color: #f65261;
  font-family: 'Montserrat';
  font-size: 20px;
}

.parent .parentCard .year {
  position: relative;
  top: 2px !important;
}

.parent .parentCard .duration {
  border-radius: 5px;
  padding: 2px 5px;
  margin-left: 15px;
}

.parent .cursor:hover {
  cursor: pointer;
}

.parent .searchButton {
  border: 0;
  border-radius: 5px;
  background-color: #f65261;
  color: #ffffff;
}

.parent .searchButton:active {
  background: #fa3142;
}
/*# sourceMappingURL=Header.module.css.map */
