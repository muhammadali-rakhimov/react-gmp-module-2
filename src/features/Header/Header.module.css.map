{
    "version": 3,
    "mappings": "AAAA,OAAO,CAAC,uEAAI;AAEZ,AAAA,OAAO,CAAC;EACN,iBAAiB,EAAE,SAAS;EAC5B,qBAAqB,EAAE,KAAK;EAC5B,eAAe,EAAE,KAAK;EACtB,gBAAgB,EAAE,OAAO;CAuH1B;;AA3HD,AAOI,OAPG,CAML,gBAAgB,CACd,SAAS,CAAC;EACR,WAAW,EAAE,wBAAwB;EACrC,SAAS,EAAE,IAAI;EACf,KAAK,EAAE,OAAO;EACd,UAAU,EAAE,sBAAsB;EAClC,MAAM,EAAE,CAAC;EACT,aAAa,EAAE,GAAG;EAClB,WAAW,EAAE,IAAI;EACjB,OAAO,EAAE,GAAG;EACZ,OAAO,EAAE,CAAC;CACX;;AAjBL,AAmBI,OAnBG,CAML,gBAAgB,CAad,SAAS,AAAA,OAAO,CAAC;EACf,UAAU,EAAE,sBAAsB;CACnC;;AArBL,AAwBE,OAxBK,CAwBL,gBAAgB,CAAC,CAAC,CAAC;EACjB,eAAe,EAAE,IAAI;EACrB,WAAW,EAAE,wBAAwB;EACrC,SAAS,EAAE,IAAI;EACf,KAAK,EAAE,OAAO;CACf;;AA7BH,AA+BE,OA/BK,CA+BL,KAAK,CAAC;EACJ,KAAK,EAAE,OAAO;EACd,WAAW,EAAE,wBAAwB;CACtC;;AAlCH,AAoCE,OApCK,CAoCL,OAAO,CAAC;EACN,OAAO,EAAE,IAAI;EACb,MAAM,EAAE,CAAC;EACT,aAAa,EAAE,GAAG;EAClB,YAAY,EAAE,IAAI;EAClB,KAAK,EAAE,GAAG;EACV,WAAW,EAAE,IAAI;EACjB,gBAAgB,EAAE,OAAO;EACzB,KAAK,EAAE,OAAO;CACf;;AAED,MAAM,CAAC,MAAM,MAAM,SAAS,EAAE,KAAK;EA/CrC,AAgDI,OAhDG,CAgDH,WAAW,CAAC;IACV,KAAK,EAAE,IAAI;IACX,UAAU,EAAE,OAAO;GACpB;EAnDL,AAqDI,OArDG,CAqDH,WAAW,CAAC;IACV,OAAO,EAAE,gBAAgB;GAK1B;EA3DL,AAwDM,OAxDC,CAqDH,WAAW,CAGT,GAAG,CAAC;IACF,aAAa,EAAE,IAAI;GACpB;;;AA1DP,AA8DE,OA9DK,CA8DL,WAAW,CAAC;EACV,WAAW,EAAE,IAAI;CA4ClB;;AA3GH,AAiEI,OAjEG,CA8DL,WAAW,CAGT,MAAM,CAAC;EACL,KAAK,EAAE,IAAI;CACZ;;AAnEL,AAqEI,OArEG,CA8DL,WAAW,CAOT,UAAU,CAAC;EACT,SAAS,EAAE,IAAI;CAChB;;AAvEL,AAyEI,OAzEG,CA8DL,WAAW,CAWT,OAAO,CAAC;EACN,MAAM,EAAE,iBAAiB;EACzB,aAAa,EAAE,IAAI;EACnB,OAAO,EAAE,kBAAkB;EAC3B,SAAS,EAAE,IAAI;EACf,WAAW,EAAE,IAAI;EACjB,KAAK,EAAE,OAAO;CACf;;AAhFL,AAkFI,OAlFG,CA8DL,WAAW,CAoBT,UAAU,CAAC;EACT,SAAS,EAAE,IAAI;EACf,KAAK,EAAE,OAAO;CACf;;AArFL,AAuFI,OAvFG,CA8DL,WAAW,CAyBT,WAAW,CAAC;EACV,WAAW,EAAE,YAAY;CAC1B;;AAzFL,AA2FI,OA3FG,CA8DL,WAAW,CA6BT,aAAa,CAAC;EACZ,KAAK,EAAE,OAAO;EACd,WAAW,EAAE,YAAY;EACzB,SAAS,EAAE,IAAI;CAChB;;AA/FL,AAiGI,OAjGG,CA8DL,WAAW,CAmCT,KAAK,CAAC;EACJ,QAAQ,EAAE,QAAQ;EAClB,GAAG,EAAE,cAAc;CACpB;;AApGL,AAsGI,OAtGG,CA8DL,WAAW,CAwCT,SAAS,CAAC;EACR,aAAa,EAAE,GAAG;EAClB,OAAO,EAAE,OAAO;EAChB,WAAW,EAAE,IAAI;CAClB;;AA1GL,AA6GE,OA7GK,CA6GL,OAAO,AAAA,MAAM,CAAC;EACZ,MAAM,EAAE,OAAO;CAChB;;AA/GH,AAiHE,OAjHK,CAiHL,aAAa,CAAC;EACZ,MAAM,EAAE,CAAC;EACT,aAAa,EAAE,GAAG;EAClB,gBAAgB,EAAE,OAAO;EACzB,KAAK,EAAE,OAAO;CACf;;AAtHH,AAwHE,OAxHK,CAwHL,aAAa,AAAA,OAAO,CAAC;EACnB,UAAU,EAAE,OAAO;CACpB",
    "sources": [
        "Header.module.scss"
    ],
    "names": [],
    "file": "Header.module.css"
}