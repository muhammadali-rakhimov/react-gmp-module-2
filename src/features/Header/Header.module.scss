@import url('https://fonts.googleapis.com/css2?family=Montserrat&display=swap');

.parent {
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-size: cover;
  background-color: #000000;

  .containerHeader {
    .addMovie {
      font-family: 'Montserrat', sans-serif;
      font-size: 18px;
      color: #f65261;
      background: rgba(96, 96, 96, 0.68);
      border: 0;
      border-radius: 4px;
      user-select: none;
      padding: 8px;
      outline: 0;
    }

    .addMovie:active {
      background: rgba(74, 74, 74, 0.68);
    }
  }

  .containerHeader a {
    text-decoration: none;
    font-family: 'Montserrat', sans-serif;
    font-size: 25px;
    color: #f65261;
  }

  .find {
    color: #ffffff;
    font-family: 'Montserrat', sans-serif;
  }

  .search {
    outline: none;
    border: 0;
    border-radius: 5px;
    margin-right: 10px;
    width: 70%;
    user-select: none;
    background-color: #424242;
    color: #ffffff;
  }

  @media screen and (max-width: 992px) {
    .imageFocus {
      width: 100%;
      object-fit: contain;
    }

    .parentCard {
      display: block !important;

      img {
        margin-bottom: 40px;
      }
    }
  }

  .parentCard {
    user-select: none;

    .image {
      width: 100%;
    }

    .movieName {
      font-size: 40px;
    }

    .rating {
      border: 1px solid #ffffff;
      border-radius: 100%;
      padding: 5px 10px 10px 10px;
      font-size: 35px;
      margin-left: 15px;
      color: #98e287;
    }

    .movieType {
      font-size: 20px;
      color: #cccccc;
    }

    .parentCard {
      font-family: 'Montserrat';
    }

    .yearDuration {
      color: #f65261;
      font-family: 'Montserrat';
      font-size: 20px;
    }

    .year {
      position: relative;
      top: 2px !important;
    }

    .duration {
      border-radius: 5px;
      padding: 2px 5px;
      margin-left: 15px;
    }
  }

  .cursor:hover {
    cursor: pointer;
  }

  .searchButton {
    border: 0;
    border-radius: 5px;
    background-color: #f65261;
    color: #ffffff;
  }

  .searchButton:active {
    background: #fa3142;
  }
}
