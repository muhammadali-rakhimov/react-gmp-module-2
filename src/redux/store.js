import { configureStore } from '@reduxjs/toolkit'
import { setupListeners } from '@reduxjs/toolkit/query'
import { rootReducer } from './reducers/rootReducer.js'
import { moviesApi } from '../services/api.js'

const store = configureStore({
  reducer: {
    root: rootReducer,
    [moviesApi.reducerPath]: moviesApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({ serializableCheck: false }).concat(
      moviesApi.middleware
    ),
})

setupListeners(store.dispatch)

export default store
