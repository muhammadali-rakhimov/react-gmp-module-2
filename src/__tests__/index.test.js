/* eslint-disable no-undef */
import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import Header from '../features/Header/Header.jsx'
import store from '../redux/store.js'

describe('App', () => {
  test('smth', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Header />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
