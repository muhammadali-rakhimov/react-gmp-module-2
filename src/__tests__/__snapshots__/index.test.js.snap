// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App smth 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <nav
      className="parent navbar navbar-light p-5"
      style={
        Object {
          "backgroundImage": "url(test-file-stub)",
        }
      }
    >
      <div
        className="container-fluid containerHeader"
      >
        <a
          className="logo"
          href="/"
        >
          <b>
            netflix
          </b>
          roulette
        </a>
        <div>
          <button
            className="addMovie"
            onClick={[Function]}
            type="button"
          >
            <span>
              + 
            </span>
            ADD MOVIE
          </button>
        </div>
      </div>
      <div
        className="find col p-5"
      >
        <div
          className="container"
        >
          <h1>
            FIND YOUR MOVIE
          </h1>
          <br />
          <form
            onSubmit={[Function]}
          >
            <input
              className="search col col-md-9 p-3"
              onChange={[Function]}
              placeholder="What do you want to watch?"
              type="text"
              value=""
            />
            <button
              className="searchButton col col-md-3 p-3"
              onClick={[Function]}
              type="submit"
            >
              SEARCH
            </button>
          </form>
        </div>
      </div>
    </nav>
  </div>
</div>
`;
